src/public/app/setup.ts

```diff
@@ -0,0 +1,130 @@
+import utils from "./services/utils.js";
+import macInit from "./services/mac_init.js";
+import ko from "knockout";
```

Comment:
There might be a problem with importing `ko`. Since it's not `import type` webpack will try to pack knockout as well and we'd end up with knockout imported twice (once in the legacy method and the second time in webpack).

I think the solution is either to try to use the types using `import type` if possible, or remove the legacy import of `ko` and apply that fix everywhere it's used.

Reply:
Knockout really is just used in the setup page and in the assets and build script
```
$ grep -lR "knockout" .
./bin/copy-dist.ts
./src/public/app/setup.ts
./src/routes/assets.ts
[…]
```
(I've redacted output to our "own" files we care about, ignoring node_modules)
Searching for "ko" also just brings up the "setup" script.

I've already removed the knockout src tag in setup.ejs (https://github.com/TriliumNext/Notes/pull/1065/commits/0559ebb48a30c198f2227fc93af0c2ff8dfc1163), so that knockout only gets pulled in via webpack already and is not imported twice.

But I agree that we can now get rid of the "manual" asset handling of it – I would want to do  the clean up in a separate PR though, if that is OK?

Reply:
ok, saw that the effort to remove the manual handling was really minimal, so I've just pushed it into this PR as well (see latest commit)